base/linalg/tridiag.jl

```diff
@@ -524,3 +548,24 @@ function A_mul_B!(C::AbstractVecOrMat, A::Tridiagonal, B::AbstractVecOrMat)
     end
     C
 end
+
+# Set-like methods
+function union!{T}(s::AbstractSet, A::Tridiagonal{T})
+    dl, d, du = A.dl, A.d, A.du
+    if length(d) <= 2
+        return union!(s,[dl;d;du])
+    else
+        return union!(s,[zero(T); dl;d;du])
+    end
+end
+function unique{T}(A::Tridiagonal{T})
+    dl, d, du = A.dl, A.d, A.du
+    # hcat operations put elements in column-major order of potential uniques
+    if length(d) <= 2
+        return unique([d[1]; dl; du ;d[end]])
+    else
+        return unique(hcat([d[1]; dl[1]; zero(T)],
```

Comment:
similarly descriptive comments here and for `SymTridiagonal` would be much appreciated